spinalcordtoolbox/reports/qc.py

```diff
@@ -264,85 +281,127 @@ def wrapped_f(sct_slice, *args):
             """
             self.qc_report.slice_name = sct_slice.get_name()
 
-            # Get the aspect ratio (height/width) based on pixel size. Consider only the first 2 slices.
-            aspect_img, self.aspect_mask = sct_slice.aspect()[:2]
-
             self.qc_report.make_content_path()
             logger.info('QcImage: %s with %s slice', func.__name__, sct_slice.get_name())
 
-            if self._angle_line is None:
-                img, mask = func(sct_slice, *args)
-            else:
-                [img, mask], centermass = func(sct_slice, *args)
+            # Get the aspect ratio (height/width) based on pixel size. Consider only the first 2 slices.
+            self.aspect_img, self.aspect_mask = sct_slice.aspect()[:2]
+
+            if self.process in ['sct_fmri_moco', 'sct_dmri_moco']:
+                [images_after_moco, images_before_moco], centermass = func(sct_slice, *args)
                 self._centermass = centermass
+                self._make_QC_image_for_4d_volumes(images_after_moco, images_before_moco)
+
+            else:
+                if self._angle_line is not None:
+                    [img, mask], centermass = func(sct_slice, *args)
+                    self._centermass = centermass
+                else:
+                    img, mask = func(sct_slice, *args)
+
+                self._make_QC_image_for_3d_volumes(img, mask, slice_orientation=sct_slice.get_name())
+
+        return wrapped_f
```

Comment:
@joshuacwnewton 
I made two functions to call in wrapped_f to make it shorter and remove all the `if self._fps is not None`. Is this the kind of changes you were expecting to make it more modular?